### PR TITLE
Updating the JS code to throw an error rather than a "string"

### PR DESCRIPTION
**Pull request by lararennie**
Changes required by clients: If anyone is doing custom behaviour depending on the value of the error, they will have to now compare to the value of e.message.

---
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1459)

### CODE REVIEW

**Comment by lararennie** [comment]
Based on original PR https://github.com/googlei18n/libphonenumber/pull/587

**Comment by andyst** [comment]
@throwing Error instead of string: Isn't that a backwards-incompatible API change?

Do we want to group it with other backwards-incompatible API changes?

**Comment by lararennie** [comment]
We are already making the next release v8.0.0 due to other backwards-incompatible changes for Java and C++.

File: tools/java/java-build/src/com/google/i18n/phonenumbers/BuildMetadataJsonFromXml.java [hunk 1]
```diff
@@ -279,6 +279,32 @@ private static void toJsArray(PhoneNumberDesc desc, JSArrayBuilder jsArrayBuilde
...
+    if (possibleLengthSize > 0) {
+      jsArrayBuilder.beginArray();
+      for (int i = 0; i < possibleLengthSize; i++) {
+        jsArrayBuilder.append(desc.possibleLength[i]);
+      }
+      jsArrayBuilder.endArray();
+    } else {
+      jsArrayBuilder.append(null);
+    }
+    // repeated int32 possible_length = 10;
```
Comment (andyst) [inline tools/java/java-build/src/com/google/i18n/phonenumbers/BuildMetadataJsonFromXml.java]: s/possible_length/possible_length_local/ ?

**Comment by ruimarinho** [comment]
Thank you for finally adding this. Refs https://github.com/googlei18n/libphonenumber/pull/690.

**Pull request merged**